### PR TITLE
feat: contact CustomizationColor

### LINKED ISSUE #18733: Issue - Accent color is not applied to the avatar placeholder in chat history and contact list
follow up for issue 2 and 3 in [#8517#issuecomment-1914867210](https://github.com/status-im/status-mobile/pull/18517#issuecomment-1914867210)

- New accent color is not applied to the avatar placeholder in chat history
- New accent color is not visible for my contact in chat

### PR DESCRIPTION
**Pull request by yqrashawn**
status-go: https://github.com/status-im/status-go/pull/4869

fixes #18733

### Summary

- status-go PR adds customization-color to contacts, contacts will now have `:customization-color`
- for initial account/contact avatar
  - replace `:background-color` with `:customization-color` in image server `get-initials-avatar-uri`
  - update `wakuext_sendContactUpdates`

https://github.com/status-im/status-mobile/assets/15090582/d00929f9-4eef-4d57-b4f9-88cff3818ce2

### Testing notes
- 

#### Platforms

- Android
- iOS

#### Areas that maybe impacted
- everything related to profile/contact customization-color
- everything related to contact
- everything related to request to join community

### Steps to test

status: ready

### CODE REVIEW

File: src/legacy/status_im/multiaccounts/update/core.cljs
Comment (seanstrom) [inline src/legacy/status_im/multiaccounts/update/core.cljs:13]: What's the meaning of passing this empty string?

File: status-go-version.json
Comment (seanstrom) [inline status-go-version.json]: Should this version number change after the PR for status-go is merged?

File: src/status_im/contexts/profile/settings/events.cljs
Comment (seanstrom) [inline src/status_im/contexts/profile/settings/events.cljs]: Similar question:
What's the meaning of the empty string?

**Review by seanstrom — COMMENTED** [review]
Hey looks good! Thank you 🙌 
I left a few small comments, lmk what you think plz 🙏

File: src/legacy/status_im/multiaccounts/update/core.cljs
Comment (yqrashawn) [reply src/legacy/status_im/multiaccounts/update/core.cljs:13]: it just means do not update profile customization color
profile color is handled by `wakuext_setCustomizationColor`

File: status-go-version.json
Comment (yqrashawn) [reply status-go-version.json]: no, I'll merge the status-go PR first after this mobile PR got approved, create a tag in status-go repo and update this version file to that tag

File: src/status_im/contexts/profile/settings/events.cljs
Comment (yqrashawn) [reply src/status_im/contexts/profile/settings/events.cljs]: Same thing, I guess `wakuext_sendContactUpdates` is actually used as set ens name? not sure about it

**Review by clauxx — APPROVED** [review]
LGTM

**Review by seanstrom — APPROVED** [review]

**Review by mohsen-ghafouri — APPROVED** [review]
💪, I think fixes should be #18733 in the description

**Review by alwx — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn, thank you for the PR. Take a look a found a critical issue. It's possible that this was caused by some commits of status-go . Hope logs might help

## Issue 1: Contact Requests Not Sent/Received

### Steps to Reproduce:
1. `UserA` goes to the profile page of `UserB`.
2. `UserA` taps the 'send contact request' button.

### Actual Result:
- The 'send contact request' button is not switched to 'sent' for `UserA` after it is tapped, until the `userB's` profile page is reopened.

https://github.com/status-im/status-mobile/assets/52490791/7bc26178-11f0-4537-ac77-2bb78f3edede

- `UserB` does not receive the Contact Request.

### Expected Result:
- The 'send contact request' button should be switched to 'sent' for UserA after it is tapped.
- UserB should receive the Contact Request.

### Logs:
[Status-debug-logs (1).zip](https://github.com/status-im/status-mobile/files/14723705/Status-debug-logs.1.zip)

**Comment by VolodLytvynenko** [comment]
## ISSUE 2: [Android] 'cannot set property “reagent-compiler2” of null' error is shown after an attempt to scan profile QR via universal scanner

### Steps:
1. `UserA` Scans the QR of `UserB`

### Actual result:
'cannot set property “reagent-compiler2” of null' error is shown

https://github.com/status-im/status-mobile/assets/52490791/6ebf1fc3-a1a6-45db-a3b5-277436f198c3


### Expected result:
The profile of `UserB` is opened

### Device:
Pixel 7a, Android 14

### Logs:
[Status-debug-logs (3).zip](https://github.com/status-im/status-mobile/files/14724875/Status-debug-logs.3.zip)

**Comment by yqrashawn** [comment]
Thanks @VolodLytvynenko, ISSUE 1 should be fixed now. ISSUE 2 is WIP

**Comment by yqrashawn** [comment]
Hi @VolodLytvynenko, I tested with local and PR build `#33` on android emulator and can't reproduce ISSUE2

Does screenshot of ISSUE 2 happen on real device?

**Comment by VolodLytvynenko** [comment]
> Hi @VolodLytvynenko, I tested with local and PR build `#33` on android emulator and can't reproduce ISSUE2
> 
> Does screenshot of ISSUE 2 happen on real device?

HI @yqrashawn yes. It was a real device Pixel 7a, Android 14

**Comment by Parveshdhull** [comment]
> ISSUE 2: [Android] 'cannot set property “reagent-compiler2” of null' error is shown after an attempt to scan profile QR via universal scanner

HI @VolodLytvynenko, Thank you for finding this issue.
In the video we are scanning Wallet account qr instead of profile QR. And I think this issue also present in develop.
Please can you confirm it and log separately.

**Comment by Parveshdhull** [comment]
Probably no need to log separately, issue is already being fixed in https://github.com/status-im/status-mobile/pull/19409
@ulisesmac please can you check video and confirm.

**Comment by VolodLytvynenko** [comment]
> Probably no need to log separately, issue is already being fixed in #19409 @ulisesmac please can you check video and confirm.

Hi @yqrashawn, yes, you are absolutely right. Thank you!
@Parveshdhull, I apologize for providing you issue with incorrect steps that is present in the develop branch. I have created it separatly

**Comment by ulisesmac** [comment]
> > Probably no need to log separately, issue is already being fixed in #19409 @ulisesmac please can you check video and confirm.
> 
> Hi @yqrashawn, yes, you are absolutely right. Thank you! @Parveshdhull, I apologize for providing you issue with incorrect steps that is present in the develop branch. I have created it separatly

The fix has been merged :eyes:

**Comment by VolodLytvynenko** [comment]
Hi @yqrashawn Thank you for PR. Take a look the found issues:
## ISSUE 2: "jump to" button on the profile page Inherits contact's color Instead of the owner's

### Steps:
1. `User A` opens the profile page of `User B`
2. Check the  "jump to" buttons

### Actual result:
"jump to" buttons inherit the color of `User B` instead of  `User A`

### Expected result:
All buttons should inherit the color of the owner, `User A`.
discussed with design https://discord.com/channels/624634427930312714/1026813762910564393/1224718951880327178

**Comment by VolodLytvynenko** [comment]
## ISSUE 3: The colors of elements in 1-1 chat are not dependent on the color preference of the receiving user

### Steps:
1. User A sets a particular color during onboarding (e.g., yellow).
2. User B opens a 1-1 chat with User A.

### Actual result:
`User B` sees the elements such as 'pin messages' and 'mute chat' sections within the chats as always blue, regardless of `User A's` selected color.


### Expected result:
`User B` should see the elements within the chats corresponding to the selected color of `User A`.

Refferencе 
https://discord.com/channels/624634427930312714/975775548007678033/1207409033225445427

**Comment by VolodLytvynenko** [comment]
## ISSUE 4: Avatar color not Immediately updated when selecting color in profile editing
### Steps to Reproduce:
1. Navigate to 'Edit Profile' -> Accent Colour.
2. Tap different color pickers and observe the avatar while tapping.

### Actual Result:
The avatar color is not changed immediately after selecting a color.

https://github.com/status-im/status-mobile/assets/52490791/bf33dd7a-cb14-407c-bc5d-f67e8b205863

### Expected Result:
The avatar color should change immediately after selecting a color

https://github.com/status-im/status-mobile/assets/52490791/9b725e13-6938-4a4c-a589-dc67444641a9

**Comment by yqrashawn** [comment]
Thanks for the review @VolodLytvynenko, ISSUE 2, 3, 4 are fixed

CleanShot 2024-04-03 at 15 12 50@2x

**Comment by VolodLytvynenko** [comment]
hi @yqrashawn Thank you for the PR. After the discussion with design it just turned out that all elements in the opened chat should also inherit the contact accent color, but this can be a follow-up. Let me know if you'll address it in this PR so I can retest. Otherwise, the PR can be merged, and I'll create a follow-up.

## ISSUE 5: Chat elements do not Inherit contact's color

### Steps:
1. `User A` sets a particular color during onboarding (e.g., yellow).
2. `User B` opens a 1-1 chat with User A.

### Actual result:
`User B` sees system icons after contact request accepting and pinned message, as well as the send button, in their default color instead of inheriting `User A's` accent color.

image

image

### Expected result:
Elements in the chat should inherit colour of the contact
Exception:
- Jump to button
- Avatar of the owner

Discussed with UI team https://discord.com/channels/624634427930312714/1224719697958670460/1225032265768374352

**Comment by yqrashawn** [comment]
let's merge it first, I got another question about pinned message icon color https://discord.com/channels/624634427930312714/1224719697958670460/1225090831971520522

**Pull request merged**
